core/src/main/java/org/wildfly/channel/spi/MavenVersionsResolver.java

```diff
@@ -55,6 +58,19 @@ public interface MavenVersionsResolver extends Closeable {
     */
    File resolveArtifact(String groupId, String artifactId, String extension, String classifier, String version) throws UnresolvedMavenArtifactException;
 
+   /**
+    * Resolve a list of maven artifacts based on the full coordinates.
+    *
+    * The order of returned Files is the same as order of coordinates.
+    *
+    * @param coordinates - list of ArtifactCoordinates. They need contain at least groupId, artifactId and version.
+    *
+    * @return a list of File representing the resolved Maven artifact.
+    *
+    * @throws UnresolvedMavenArtifactException if th artifact can not be resolved.
```

Comment:
should be?

`@throws UnresolvedMavenArtifactException if any artifact can not be resolved.`